client/layout/transient-notices/index.js

```diff
@@ -23,51 +22,51 @@ function TransientNotices( props ) {
 		createNotice: createNotice2,
 		removeNotice: onRemove2,
 	} = useDispatch( 'core/notices2' );
-	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { dismissNotice } = useDispatch( NOTICES_STORE_NAME );
 	const {
-		currentUser = {},
+		hasFinishedResolution,
 		notices = [],
 		notices2 = [],
-		noticesQueue = {},
+		noticesQueue = [],
 	} = useSelect( ( select ) => {
+		const currentUser = select( USER_STORE_NAME ).getCurrentUser();
+
 		// NOTE: This uses core/notices2, if this file is copied back upstream
 		// to Gutenberg this needs to be changed back to just core/notices.
 		return {
-			currentUser: select( USER_STORE_NAME ).getCurrentUser(),
+			currentUser,
+			hasFinishedResolution: select(
+				NOTICES_STORE_NAME
+			).hasFinishedResolution( 'getNotices', [ currentUser.id ] ),
 			notices: select( 'core/notices' ).getNotices(),
 			notices2: select( 'core/notices2' ).getNotices(),
-			noticesQueue: select( OPTIONS_STORE_NAME ).getOption(
-				QUEUE_OPTION
-			),
+			noticesQueue:
+				currentUser && currentUser.id
+					? Object.values(
+							select( NOTICES_STORE_NAME ).getNotices(
+								currentUser.id
+							)
+					  )
+					: [],
 		};
 	} );
 
-	const getCurrentUserNotices = () => {
-		return Object.values( noticesQueue ).filter(
-			( notice ) => notice.user_id === currentUser.id || ! notice.user_id
-		);
-	};
-
-	const dequeueNotice = ( id ) => {
-		const remainingNotices = { ...noticesQueue };
-		delete remainingNotices[ id ];
-		updateOptions( {
-			[ QUEUE_OPTION ]: remainingNotices,
-		} );
-	};
-
 	useEffect( () => {
-		getCurrentUserNotices().forEach( ( queuedNotice ) => {
+		if ( ! hasFinishedResolution ) {
+			return;
+		}
+
+		noticesQueue.forEach( ( queuedNotice ) => {
 			const notice = applyFilters( QUEUED_NOTICE_FILTER, queuedNotice );
 
 			createNotice2( notice.status, notice.content, {
```

Comment:
Could this logic be moved to the data store potentially? Given we could trigger an action from a resolver.

Reply:
That's an interesting idea, @louwie17.  I guess the question is whether or not we would ever want the ability to resolve/get the notices without first displaying them.